indexing-service/src/test/java/io/druid/indexing/overlord/http/OverlordResourceTest.java

```diff
@@ -0,0 +1,305 @@
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.retry.RetryOneTime;
+import org.apache.curator.test.TestingServer;
+import org.apache.curator.test.Timing;
+import org.easymock.EasyMock;
+import org.joda.time.Period;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import javax.ws.rs.core.Response;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+public class OverlordResourceTest
+{
+  private TestingServer server;
+  private Timing timing;
+  private CuratorFramework curator;
+  private TaskMaster taskMaster;
+  private TaskLockbox taskLockbox;
+  private TaskStorage taskStorage;
+  private TaskActionClientFactory taskActionClientFactory;
+  private CountDownLatch announcementLatch;
+  private DruidNode druidNode;
+  private OverlordResource overlordResource;
+  private CountDownLatch[] taskCountDownLatches;
+
+  private void setupServerAndCurator() throws Exception
+  {
+    server = new TestingServer();
+    timing = new Timing();
+    curator = CuratorFrameworkFactory
+        .builder()
+        .connectString(server.getConnectString())
+        .sessionTimeoutMs(timing.session())
+        .connectionTimeoutMs(timing.connection())
+        .retryPolicy(new RetryOneTime(1))
+        .compressionProvider(new PotentiallyGzippedCompressionProvider(true))
+        .build();
+  }
+
+  private void tearDownServerAndCurator()
+  {
+    CloseQuietly.close(curator);
+    CloseQuietly.close(server);
+  }
+
+  @Before
+  public void setUp() throws Exception
+  {
+    taskLockbox = EasyMock.createStrictMock(TaskLockbox.class);
+    taskLockbox.syncFromStorage();
+    EasyMock.expectLastCall().atLeastOnce();
+    taskLockbox.unlock(EasyMock.<Task>anyObject());
+    EasyMock.expectLastCall().atLeastOnce();
+    taskActionClientFactory = EasyMock.createStrictMock(TaskActionClientFactory.class);
+    EasyMock.expect(taskActionClientFactory.create(EasyMock.<Task>anyObject()))
+            .andReturn(null).anyTimes();
+    EasyMock.replay(taskLockbox, taskActionClientFactory);
+
+    taskStorage = new HeapMemoryTaskStorage(new TaskStorageConfig(null));
+    taskCountDownLatches = new CountDownLatch[2];
+    taskCountDownLatches[0] = new CountDownLatch(1);
+    taskCountDownLatches[1] = new CountDownLatch(1);
+    announcementLatch = new CountDownLatch(1);
+    IndexerZkConfig indexerZkConfig = new IndexerZkConfig(new ZkPathsConfig(), null, null, null, null, null);
+    setupServerAndCurator();
+    curator.start();
+    curator.create().creatingParentsIfNeeded().forPath(indexerZkConfig.getLeaderLatchPath());
+    druidNode = new DruidNode("hey", "what", 1234);
+    ServiceEmitter serviceEmitter = new ServiceEmitter(null, null, null){
```

Comment:
NoopServiceEmitter has test scope. I can put it out in main folder or I can declare dependency on the test-jar which has this class or I can keep it as it is. What do you think ?